### PR TITLE
[SDK-4546] Add orgs in client credentials support

### PR DESCRIPTION
**Pull request by adamjmcgrath**
### Changes

- Add optional organization parameter to client credentials grant
- Add Management API endpoints to /client-grants and /organizations

### CODE REVIEW

File: lib/auth0/api/v2/organizations.rb [hunk 1]
```diff
@@ -330,6 +330,52 @@ def delete_organizations_member_roles(organization_id, user_id, roles = [])
...
+        def get_organizations_client_grants(organization_id, options= {})
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_id' if organization_id.to_s.empty?
+          request_params = {
+            client_id:      options.fetch(:client_id, nil),
+            audience:       options.fetch(:audience, nil),
+            per_page:       options.fetch(:per_page, nil),
+            page:           options.fetch(:page, nil),
+            include_totals: options.fetch(:include_totals, nil)
+          }
+          path = "#{organizations_client_grants_path(organization_id)}"
```
Comment (Widcket) [inline lib/auth0/api/v2/organizations.rb:350]: Is it necessary to interpolate the return value, if `organizations_client_grants_path(organization_id)` returns a string?

File: lib/auth0/api/v2/organizations.rb [hunk 2]
```diff
@@ -330,6 +330,52 @@ def delete_organizations_member_roles(organization_id, user_id, roles = [])
...
+          get(path, request_params)
+        end
+
+        # Associate a client grant with an organization
+        # @param organization_id [string] The Organization ID
+        # @param grant_id [string] The Client Grant ID you want to associate to the Organization.
+        def create_organizations_client_grant(organization_id, grant_id)
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_id' if organization_id.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid grant_id' if grant_id.empty?
```
Comment (Widcket) [inline lib/auth0/api/v2/organizations.rb:359]:
```suggestion
          raise Auth0::InvalidParameter, 'Must supply a valid grant_id' if grant_id.to_s.empty?
```

File: lib/auth0/api/v2/organizations.rb [hunk 3]
```diff
@@ -330,6 +330,52 @@ def delete_organizations_member_roles(organization_id, user_id, roles = [])
...
+          path = "#{organizations_client_grants_path(organization_id)}"
+          post(path, body)
+        end
+
+        # Remove a client grant from an organization
+        # @param organization_id [string] The Organization ID
+        # @param grant_id [string] The Client Grant ID you want to remove from the Organization.
+        def delete_organizations_client_grant(organization_id, grant_id)
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_id' if organization_id.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid grant_id' if grant_id.empty?
```
Comment (Widcket) [inline lib/auth0/api/v2/organizations.rb:373]:
```suggestion
          raise Auth0::InvalidParameter, 'Must supply a valid grant_id' if grant_id.to_s.empty?
```

File: lib/auth0/api/v2/organizations.rb [hunk 4]
```diff
@@ -330,6 +330,52 @@ def delete_organizations_member_roles(organization_id, user_id, roles = [])
...
+        # @param organization_id [string] The Organization ID
+        # @param grant_id [string] The Client Grant ID you want to associate to the Organization.
+        def create_organizations_client_grant(organization_id, grant_id)
+          raise Auth0::MissingOrganizationId, 'Must supply a valid organization_id' if organization_id.to_s.empty?
+          raise Auth0::InvalidParameter, 'Must supply a valid grant_id' if grant_id.empty?
+          
+          body = {}
+          body[:grant_id] = grant_id
+
+          path = "#{organizations_client_grants_path(organization_id)}"
```
Comment (Widcket) [inline lib/auth0/api/v2/organizations.rb:364]: Same as before, is the string interpolation necessary?

File: lib/auth0/api/v2/client_grants.rb [hunk 5]
```diff
@@ -11,13 +11,15 @@ module ClientGrants
         # @param audience [string] The audience of the client grant to retrieve.
         # @param page [int] Page number to get, 0-based.
         # @param per_page [int] Results per page if also passing a page number.
+        # @param allow_any_organization [bool] Optional filter on allow_any_organization.
```
Comment (Widcket) [inline lib/auth0/api/v2/client_grants.rb:14]: Should we expand a bit on what `allow_any_organization` does?

File: spec/lib/auth0/api/v2/organizations_spec.rb [hunk 6]
```diff
@@ -639,4 +639,70 @@
...
+      expect(@instance).to receive(:post).with(
+        '/api/v2/organizations/org_id/client-grants', {
+        grant_id: 'grant_id'
+      })
+      expect { @instance.create_organizations_client_grant('org_id', 'grant_id') }.not_to raise_error
+    end
+  end
+
+  context '.delete_organizations_client_grants' do
+    it 'is expected to respond to a delete_organizations_client_grants method' do
```
Comment (Widcket) [inline spec/lib/auth0/api/v2/organizations_spec.rb:697]:
```suggestion
  context '.delete_organizations_client_grant' do
    it 'is expected to respond to a delete_organizations_client_grant method' do
```

File: lib/auth0/api/v2/organizations.rb (hunk 1 above)
Comment (adamjmcgrath) [reply lib/auth0/api/v2/organizations.rb:350]: Maybe not? Am just being consistent with the rest of the file https://github.com/auth0/ruby-auth0/pull/540/files/43bb301afee4d5ebd923eb897b53b998bc3465b6#diff-3936cbc470adbe60f39a627f01d4259b0c1121de790b4e7d3144c536f073ccf7R324

File: lib/auth0/api/v2/organizations.rb (hunk 4 above)
Comment (adamjmcgrath) [reply lib/auth0/api/v2/organizations.rb:364]: Same as before, just being consistent...

File: lib/auth0/api/v2/client_grants.rb (hunk 5 above)
Comment (adamjmcgrath) [reply lib/auth0/api/v2/client_grants.rb:14]: I'd agree if you were setting the `allow_any_organization` field, but this is a filter on the value

**Review by Widcket — APPROVED** [review]

**Pull request merged**